File: apps/web/src/api/agent.ts
import { apiClient } from './client';
import type { ApiResponse } from './types';

export type AgentRunRequest = {
  input: string;
  context?: Record<string, unknown>;
};

export type AgentStepResponse = {
  name: string;
  status: string;
  detail: string;
};

export type AgentRunResponse = {
  requestId: string;
  agentCode: string;
  intent: string;
  answer: string;
  steps: AgentStepResponse[];
  metadata?: Record<string, unknown>;
  latencyMs: number;
};

export const agentApi = {
  runDemo: async (request: AgentRunRequest) => {
    const response = await apiClient.post<ApiResponse<AgentRunResponse>>('/agents/demo/run', request);
    return response.data.data;
  },
};

File: apps/agent/app/schemas/agent.py
from typing import Any

from pydantic import BaseModel, Field


class AgentRunRequest(BaseModel):
    input: str = Field(..., min_length=1, max_length=4000)
    tenantId: str | None = None
    userId: str | None = None
    requestId: str | None = None
    context: dict[str, Any] = Field(default_factory=dict)


class AgentStep(BaseModel):
    name: str
    status: str
    detail: str


class AgentRunResponse(BaseModel):
    requestId: str
    agentCode: str
    intent: str
    answer: str
    steps: list[AgentStep]
    metadata: dict[str, Any] = Field(default_factory=dict)
    latencyMs: int
